src/main/components/Mockup.jsx

```diff
@@ -0,0 +1,38 @@
+import React, { useState } from 'react';
+import NameForm from './NameForm.jsx';
+import MockupName from './MockupName.jsx';
+import DataTree from './DataTree.jsx';
+
+const dataTreeOptions = {
+  onAdd: true,
+  onEdit: true,
+  onDelete: true,
+  enableClipboard: false,
+};
+
+const Mockup = (props) => {
+  const { test, index } = props;
+
+  // State
+  const [name, setName] = useState(test.name);
+
+  // Mode controls display of name vs. name edit form, and is set by child compoments
+  const [mode, setMode] = useState('view');
```

Comment:
Not a big deal, and I might be under-thinking it, but could we use `(name === '')` instead of a new piece of state?  Not complaining, just a thought as we continue to touch up-- clearly a lot of good work here for a feature you rightly thought would boost the professional feel and value of the product

Reply:
The main reason I put `name` in state was to force a re-render. If we just update the name in the tests array, it doesn’t change state so the new name doesn’t get displayed until something else forces a re-render.